open-company/open-company-web
# Improve initial load

#### Review after #927 


Try to delay the loading of all the resources that are not immediately needed.
It delays by 2.5 seconds:
- all the sections both sorts except the one currently selected
- AP both sorts if it's not currently selected
- team, roster and channels: delay only if the app is not loaded with the one of the settings panel open, also roster and channels call where particularly expensive for Slack because they include a sync call to Slack apis

To test:
- open staging AP (with dev tools closed)
- now refresh and time how long it take to show the posts
- same for a section (the one with more posts)
- now deploy this branch
- repeat the timing for the 3 urls above
- [x] did it improve?
- [x] now after loading AP or a section make sure the settings panels are all working
- [x] now load `/carrot/all-posts?org-settings=org` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?org-settings=integrations` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?org-settings=team` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?org-settings=invite` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?user-settings=profile` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?user-settings=notifications` and make sure the settings panel is loaded as expected
- [x] now load `/carrot/all-posts?reminders=true` and make sure the settings panel is loaded as expected
